### PR TITLE
VM Import: Use datastore name when datastore path is not set to search for pools on DB

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR addresses an issue during VM import on VMFS storage, storage pool cannot be found during import operation and it fails:

````
2024-03-08 00:49:57,086 DEBUG [c.c.s.d.GuestOSHypervisorDaoImpl] (API-Job-Executor-1:ctx-8d3d66e6 job-37 ctx-2c2f3dfe) (logid:0dfa2610) Mapping for guest OS ID: windows2019srvNext_64Guest for hypervisor: VMware with version: 8.0.0.2 can not be found. Trying to find one for the parent version: 8.0
2024-03-08 00:49:57,099 DEBUG [o.a.c.v.UnmanagedVMsManagerImpl] (API-Job-Executor-1:ctx-8d3d66e6 job-37 ctx-2c2f3dfe) (logid:0dfa2610) Trying to import VM [{"name":"Mig01","internalCSName":"Mig01","powerState":"PowerOn","cpuCores":8,"cpuCoresPerSocket":8,"memory":32768,"cpuSpeed":0,"operatingSystemId":"windows2019srvNext_64Guest","operatingSystem":"Microsoft Windows Server 2022 (64-bit)","clusterName":"CLS01","hostName":"esx01.flexapp360.cloud","disks":[{"diskId":"5788-2000","label":"Hard disk 1","capacity":429496729600,"fileBaseName":"Mig01","imagePath":"[FA360-DS01-ESX] Mig01/Mig01.vmdk","controller":"lsisas1068","controllerUnit":0,"position":0,"chainInfo":"{\"diskDeviceBusName\":\"scsi0:0\",\"diskChain\":[\"[FA360-DS01-ESX] Mig01/Mig01.vmdk\"]}","datastoreName":"FA360-DS01-ESX","datastorePort":0}],"nics":[{"nicId":"Network adapter 1","adapterType":"E1000","macAddress":"00:50:56:bd:3b:3d","vlan":101,"pciSlot":"com.vmware.vim25.VirtualDevicePciBusSlotInfo@6821ea33"}]}] with name ["Mig01"], in zone [error decoding Zone {"id": "1", "name": "ZONE01", "uuid": "ce2b79a1-098c-44c0-a0d7-82b18fb74085"}], cluster [error decoding Cluster {id: "1", name: "vc01.flexapp360.cloud/DC01/CLS01", uuid: "2371d542-fec0-41ca-b7c9-56d14fd13dd0"}], and host [error decoding Host {"id":1,"name":"esx01.flexapp360.cloud","type":"Routing","uuid":"52f61c1e-f1e3-45ef-9c71-3f01b791798c"}], using template [error decoding Template {"format":"ISO","id":202,"uniqueName":"system-default-vm-import-dummy-template.iso"}], service offering [error decoding Service offering {"id":14,"name":"Custom","uuid":"9732133c-f7b5-4b4e-a5f8-b92e913a74ad"}.], disks map [{}], NICs map [{"Network adapter 1":204}] and details [{"cpuNumber":"8","memory":"32768","cpuSpeed":"1000"}].
2024-03-08 00:49:57,102 DEBUG [c.c.u.AccountManagerImpl] (API-Job-Executor-1:ctx-8d3d66e6 job-37 ctx-2c2f3dfe) (logid:0dfa2610) Access granted to Account [{"accountName":"admin","id":2,"uuid":"82119e65-dcc4-11ee-a861-0050568d528d"}] to Service offering {"id":14,"name":"Custom","uuid":"9732133c-f7b5-4b4e-a5f8-b92e913a74ad"}. by AffinityGroupAccessChecker
2024-03-08 00:49:57,124 DEBUG [o.a.c.f.j.i.AsyncJobManagerImpl] (API-Job-Executor-1:ctx-8d3d66e6 job-37) (logid:0dfa2610) Complete async job-37, jobStatus: FAILED, resultCode: 530, result: org.apache.cloudstack.api.response.ExceptionResponse/null/{"uuidList":[],"errorcode":"530","errortext":"Storage pool for disk Hard disk 1(5788-2000) with datastore: FA360-DS01-ESX not found in zone ID: ce2b79a1-098c-44c0-a0d7-82b18fb74085"}
````
 
Fixes: #8712 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

File: server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java [hunk 1]
```diff
@@ -551,7 +551,8 @@ private StoragePool getStoragePool(final UnmanagedInstanceTO.Disk disk, final Da
             List<StoragePoolVO> pools = primaryDataStoreDao.listPoolsByCluster(cluster.getId());
             pools.addAll(primaryDataStoreDao.listByDataCenterId(zone.getId()));
             for (StoragePool pool : pools) {
-                if (StringUtils.contains(pool.getPath(), dsPath)) {
+                String searchPoolParam = StringUtils.isNotBlank(dsPath) ? dsPath : dsName;
```
Comment (sureshanaparti) [inline server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java:554]:
```suggestion
                String searchPoolPath = StringUtils.isNotBlank(dsPath) ? dsPath : dsName;
```

**Review by sureshanaparti — APPROVED** [review]
minor nit, code LGTM

**Review by slavkap — APPROVED** [review]
code LGTM, tested with NFS/StorPool on KVM

**Pull request merged**